packages/react-forms/src/InputDate/index.js

```diff
@@ -21,141 +22,144 @@ function toYYYYMMDD(date) {
-  };
-
-  preventDefault = (evt: Event) => evt.preventDefault();
-
-  render() {
-    const {
-      value,
-      onChange, // eslint-disable-line no-unused-vars
-      isOpen: isOpenProp, // eslint-disable-line no-unused-vars
-      onToggle, // eslint-disable-line no-unused-vars
-      min,
-      max,
-      ...props
-    } = this.props;
-
-    const { current } = this.state;
-
-    const isOpen = this.isOpen;
-
-    const dateValue = parse(value);
-    const isDateValid = !isNaN(dateValue.getTime());
-
-    return (
-      <MouseOutside
-        onClickOutside={this.handleClose}
-        refs={[this.refA, this.refB]}
-      >
-        {() => (
-          <Manager>
-            <Reference>
-              {({ ref }) => (
-                <InputText
-                  type="date"
-                  renderAddon={({ marginRightClass }) => (
-                    <Icon
-                      name="calendar"
-                      className={marginRightClass}
-                      onClick={!props.disabled ? this.handleOpen : undefined}
-                    />
-                  )}
-                  onFocus={this.handleOpen}
-                  focus={isOpen}
-                  onBlur={this.handleCloseOnBlur}
-                  value={value}
-                  onChange={this.handleInputChange}
-                  onClick={this.preventDefault}
-                  min={min}
-                  max={max}
-                  ref={mergeRefs(ref, this.refA)}
-                  {...props}
-                />
-              )}
-            </Reference>
-            {isOpen && (
-              <Popper placement="bottom">
-                {({ ref, style }) => (
-                  <Calendar
-                    onChangeCurrent={current => this.setState({ current })}
-                    current={current || (isDateValid ? dateValue : new Date())}
-                    onSelect={this.handleSelect}
-                    selected={isDateValid ? dateValue : undefined}
-                    minDate={min ? new Date(min) : undefined}
-                    maxDate={max ? new Date(max) : undefined}
-                    ref={mergeRefs(ref, this.component, this.refB)}
-                    style={style}
+const InputDate = ({
+  value,
+  onChange,
+  defaultIsOpen,
+  isOpen: isOpenProp,
+  onToggle,
+  min,
+  max,
+  onCalendarChange,
+  calendarValue,
+  ...props
+}: Props) => {
+  const inputRef = createRef();
+  const calendarRef = createRef();
+
+  const calendarDate = calendarValue == null ? undefined : parse(calendarValue);
+  const dateValue = parse(value);
+  const isDateValid = !isNaN(dateValue.getTime());
+
+  const [isOpen = false, setOpen] = useControlledState(
+    defaultIsOpen,
+    isOpenProp,
+    onToggle
+  );
+
+  const [baseCurrent, setCurrent] = useControlledState(
+    undefined,
+    calendarDate,
+    date => onCalendarChange && onCalendarChange(toYYYYMMDD(date))
+  );
+
+  const current = baseCurrent || dateValue;
```

Comment:
@FezVrasta You removed this, I am not sure why but it's still needed. Also note line 64.

Reply:
I did replace it with

```
const [current = dateValue, setCurrent] = useControlledState(
```

which should do the same thing but it's cleaner. Maybe I accidentally removed this as well?

Reply:
You changed it this way:
```
const [current, setCurrent] = useControlledState(
    dateValue,
    calendarValue,
    onCalendarChange
  );
```
I propose this then:
```
const [current = dateValue, setCurrent] = useControlledState(
    undefined,
    calendarDate....
```

At the same time I'll refactor `onCalendarChange`